nomenclature/codelist.py

```diff
@@ -364,6 +335,37 @@ def read_excel(cls, name, source, sheet_name, col, attrs=None):
 
         return cls(name=name, mapping=mapp)
 
+    def check_illegal_characters(self, config: NomenclatureConfig) -> Dict[str, Code]:
+        """Check that no illegal characters are left in codes after tag replacement"""
+        illegal = (
+            ["{", "}"] + config.illegal_characters
+            if config.check_illegal_characters
+            else ["{", "}"]
+        )
+
+        def _check_string(value):
+            if isinstance(value, str):
+                if any(char in value for char in illegal):
+                    raise ValueError(
+                        f"Unexpected character in {self.name}: '{code.name}'."
+                        " Check for illegal characters and/or if tags were spelled correctly."
+                    )
+            elif isinstance(value, dict):
+                for k in value.keys():
+                    _check_string(k)
+                for v in value.values():
+                    _check_string(v)
+            elif isinstance(value, list):
+                for item in value:
+                    _check_string(item)
+
+        for code in self.mapping.values():
+            if not code.repository:
+                for attr in code.model_fields:
```

Comment:
You can simplify this by using the `model_dump` method provided by pydantic (https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_dump):
```python
                for value in code.model_dump(exclude="file").values():
                    _check_string(value)
```
This way has two benefits, firstly you don't have to check if the current attribute is a file and secondly you don't have to explicitly fetch the value.